doc/doxygen/src/kconfig/kconfig.md

```diff
@@ -0,0 +1,366 @@
+
+The main `Kconfig` file is used in this step to show the configurable
+parameters of the system. Kconfig will filter not available parameters (i.e.
+parameters exposed by modules that are not being used) based on the file
+`$ (GENERATED_DIR)/Kconfig.dep` generated in step 1.
+
+By default, during the transition phase module configuration via Kconfig will
+not be enabled, and the user should explicitly enable this (see
+[Making configuration via Kconfig optional](#kconfig-configuration-optional)).
+
+`merged.config` is one of the inputs for menuconfig. This means that any
+configuration that the application defines in the `app.config` or a backup
+configuration from the user in `user.config` are taken into account on the
+first run (see [Appendix C](#kconfig-appendix-c)).
+
+In this step the user chooses configuration values (or selects the minimal
+configuration) and saves it to the `merged.config` file. Here the user can
+choose to save a backup configuration file for later at a different location
+(e.g. a `user.config` file in the application folder).
+
+#### Input
+- `/Kconfig` file.
+- Optional:
+    - `$ (APPDIR)/app.config`
+    - `$ (APPDIR)/user.config`
+    - `$ (GENERATED_DIR)/merged.config`
+
+#### Output
+- Updated `$ (GENERATED_DIR)/merged.config` file.
+- `$ (GENERATED_DIR)/merged.config.old` backup file.
+
+### 4. Generation of the autoconf.h header
+With the addition of Kconfig a dependency has been added to the build
+process: the `$ (GENERATED_DIR)/autoconf.h` header file. This header file is
+the main output from the Kconfig configuration system. It holds all the macros
+that should be used to configure modules in RIOT:
+`CONFIG_<module>_<parameter>`.
+
+In order to generate the `autoconf.h` file the `genconfig` script is used.
+Inputs for this script are the main `Kconfig` file and `merged.config`
+configuration file, which holds the selected values for the exposed parameters.
+
+#### Input:
+- `$ (GENERATED_DIR)/merged.config` file.
+- Main `Kconfig` file exposing configuration of modules.
+
+#### Output:
+- `$ (GENERATED_DIR)/autoconf.h` configuration header file.
+
+
+### Summary of files
+These files are defined in `kconfig.mk`.
+
+| File              | Description |
+| ----------------- | ----------- |
+| `Kconfig`           | Defines configuration options of modules. |
+| `Kconfig.dep`       | Holds a list of the modules that are being compiled. |
+| `app.config`        | Holds default application configuration values. |
+| `user.config`       | Holds configuration values applied by the user. |
+| `merged.config`     | Holds configuration from multiple sources. Used to generate header. |
+| `autoconf.h`        | Header file containing the macros that applied the selected configuration. |
+
+## Kconfig symbols in Makefiles
+As '.config' files have Makefile syntax they can be included when building.
+This allows to access the applied configuration from the build system. This
```

Comment:
Make it one sentence to avoid duplicate sentences with "this": This allows .....  system **which ** can check for enabled modules in future.